src/main/java/ch/njol/skript/expressions/arithmetic/ExprArithmetic.java

```diff
@@ -304,6 +307,10 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 		}
 
 		arithmeticGettable = ArithmeticChain.parse(chain);
+		if (returnType == Object.class) // get everything it might be
+			knownReturnTypes = Arithmetics.getAllReturnTypes(operator);
+		else
+			knownReturnTypes = Collections.emptySet();
```

Comment:
We can get the possible return types from [here](https://github.com/SkriptLang/Skript/pull/6624/files#diff-51bffaa9a4d59841ea60812f2a81fc4bcb7d95da16a1dcb5760a38bebd151498R241) instead